explorer/mempool.go

```diff
@@ -230,14 +242,38 @@ func (exp *explorerUI) storeMempoolInfo() (lastBlockHash string, lastBlock int64
 		LatestTransactions: latest,
 		FormattedTotalSize: humanize.Bytes(uint64(totalSize)),
 		TicketIndexes:      txindexes,
+		VotingInfo:         votingInfo,
 	}
 	return
 }
 
+// addVoteIndex sets v.MempoolTicketIndex to the corresponding map value if it exits in
+// txindexes and creates a new index map if it doesn't
+func addVoteIndex(v *VoteInfo, txindexes map[int64]map[string]int) {
+	if idxs, ok := txindexes[v.Validation.Height]; ok {
+		if idx, ok := idxs[v.TicketSpent]; ok {
+			v.MempoolTicketIndex = idx
+		} else {
+			idx := len(idxs) + 1
+			idxs[v.TicketSpent] = idx
+			v.MempoolTicketIndex = idx
+		}
+	} else {
+		idxs := make(map[string]int)
+		idxs[v.TicketSpent] = 1
+		txindexes[v.Validation.Height] = idxs
+		v.MempoolTicketIndex = 1
+	}
+}
+
 func voteForLastBlock(blockHash, validationHash string) bool {
 	return blockHash == validationHash && blockHash != ""
 }
 
+func updateBlockValidity(votingInfo *VotingInfo) {
+	votingInfo.BlockValid = votingInfo.Valids+votingInfo.Invalids >= votingInfo.TotalNeeded && votingInfo.Valids >= votingInfo.Required
```

Comment:
I see your point.  I reviewed DCRD briefly and it checks a lot of things that would likely require quite a few rpc calls to replicate if we really want to say a block is valid.  I recommend one of three paths here:

1.  Remove "Vote Choices" completely from the mempool page.  Since the votes seem to come in via ws all at once anyway after a block is sent, we don't really get useful information out of this (IE, we don't actually see votes come in one at a time allowing us to watch the block become "valid")
2.  Remove the assertion of validity here and change it to be something like:
VOTES:  3 found / 5 targeted
3.  tone down the assertion of validity:
VOTES:  3 found / 5 targeted (vote quantity valid)

Reply:
@chappjc or @raedah any feedback on which option to take here?

Reply:
I generally agree with 1.  Votes come all at once (within a window of a couple seconds at most), so it's not so useful in the new tx loop.  However, if done after a few second delay, it perhaps makes sense.

But say we wanted to write the validity check correctly, I would not favor RPC calls (which ones, by the way?).  But thinking about options 2. and 3., I do think it is wise not to jump to conclusions about validity until at least 3 up/down votes are received.  We know that 2 up + 2 down = invalid/disapprove, but until a third comes in or the next block is mined, it's technically up in the air.

So I see this not as a boolean but

```go
type BlockValidState int8
const (
	BlockValidYes BlockValidState = iota
	BlockValidNo
	BlockValidUncertain
)
```

Reply:
Throwing this out there as I'm not familiar with the details:
A.  miners create what they think is a valid block
B.  voters validate the block by running that through a bunch of checks (dcrd) in maybeacceptblock and then even more in checkConnectBlock :

https://github.com/decred/dcrd/blob/9536f0c88fa85180c348043766f6c1d4d42107da/blockchain/accept.go#L104-L116

https://github.com/decred/dcrd/blob/9536f0c88fa85180c348043766f6c1d4d42107da/blockchain/validate.go#L2305-L2323

Items confirmed would be things like:
1.  previous block is valid
2.  check block header
3.  check block max size
4.  check that all transactions are finalized in block
5.  confirm coinbase contains block height
6.  confirm votes are for winning tickets
7.  check for duplicate transactions
8.  check block scripts
9.  check that there is at least 1 regular transaction
10.  check that there is only one coinbase transaction
11.  block header commits to votes and tickets and revocations

I'm sure there is more, and I'm sure this is a partial list, but thats what I saw when I was looking at replicating block validity from dcrd into dcrdata.  

Shall I implement #1 here and remove the reference to validity so we can close out this PR?

Reply:
We're only concerned with B.  This is actually why some people, @davecgh for one if I recall correctly, have have emphasized that the voters *approve* or *disapprove*.  Strictly speaking, validation is what you've described primarily in your last comment.  We can assume that dcrd has done its job, and that it is a valid block.

This is mempool information, so all we want to do it consider the votes that are likely to be included in the next block, which approve the current best block, and indicate how the block would be treated (approved or disapproved).

Anyway, yes, I say we not provide an indication of the collective vote choice on the mempool page.  Besides the implementation and timing issues, it's really not certain until a miner includes the votes in a block, which they really don't have to do (although they are incentivized to include all votes).